wled00/presets.cpp

```diff
@@ -120,6 +120,15 @@ bool applyPreset(byte index, byte callMode)
   return true;
 }
 
+// apply preset or fallback to a effect and palette if it doesn't exist
+void presetFallback(uint8_t presetID, uint8_t effectID, uint8_t paletteID)
+{
+  applyPreset(presetID, CALL_MODE_BUTTON_PRESET);
```

Comment:
IR is simulating button presses, moving function here is not simulating buttons. The CALL_MODE_BUTTON_PRESET is not appropriate in such case and should be adjusted when calling `presetFallback()`.

Reply:
Makes total sense. Will also update to match the call signature from the other method more closely to make it easier to understand by inspection

Reply:
Looks like I was not clear enough. Sorry for that.

My thought was to just duplicate (or reuse) existing function from ir.cpp *or* make a universal function that may be usable in plethora of other situations (future-proof) by adding a call mode parameter instead of hard-coded CALL_MODE_BUTTON_PRESET.

Reply:
If I'm understanding you correctly I think you were clear and my change was to take the latter approach based on your feedback. 

I added a universal version of the [preset with fallback](https://github.com/Aircoookie/WLED/pull/3237/files#diff-2659ccd7699e79b87d515cd0b1676e2a25c7dd1959fafa352620805b2b4e8ed0R124) function that accepts a `callMode`, I just changed the signature a bit so that the new method name (`applyPresetWithFallback`) and argument order would more closely match the existing universal `applyPreset` method compared to my initial code.

Apologies if I'm still misinterpreting, though, and happy to do another pass.

Reply:
Indeed. My bad.
You can remove the 3 lines in ir.cpp that are no longer needed in `presetFalback()` or entirely remove `presetFalback()` replacing other code as necessary.